### PR TITLE
fixed region definition for RZ designs

### PR DESCRIPTION
**Pull request by IreneWoyna**

### CODE REVIEW

File: pyaedt/modeler/cad/Primitives2D.py [hunk 1]
```diff
@@ -288,7 +288,8 @@ def create_region(self, pad_percent=300, is_percentage=True):
         ----------
         pad_percent : float, str, list of floats or list of str, optional
             Same padding is applied if not a list. The default is ``300``.
-            If a list of floats or str, interpret as adding for ``["+X", "+Y", "-X", "-Y"]``.
+            If a list of floats or str, interpret as adding for ``["+X", "+Y", "-X", "-Y"]`` for XY geometry mode,
```
Comment (MaxJPRey) [inline pyaedt/modeler/cad/Primitives2D.py:291]:
```suggestion
            If a list of floats or str, interpret as adding ``["+X", "+Y", "-X", "-Y"]`` for XY geometry mode,
```

File: pyaedt/modeler/cad/Primitives2D.py [hunk 2]
```diff
@@ -304,9 +305,29 @@ def create_region(self, pad_percent=300, is_percentage=True):
         """
         if not isinstance(pad_percent, list):
```
Comment (SMoraisAnsys) [inline pyaedt/modeler/cad/Primitives2D.py:306]:
```suggestion
        """
        SOLUTION_TYPES = ["TransientXY", "MagnetostaticXY", "EddyCurrentXY", "ElectrostaticXY", "DCConductionXY", "ACConductionXY"]
        if not isinstance(pad_percent, list):
```

File: pyaedt/modeler/cad/Primitives2D.py [hunk 3]
```diff
@@ -304,9 +305,29 @@ def create_region(self, pad_percent=300, is_percentage=True):
...
             if self._app.design_type == "2D Extractor" or self._app.design_type == "Maxwell 2D":
-                pad_percent = [pad_percent, pad_percent, 0, pad_percent, pad_percent, 0]
+                if self._app.solution_type in [
+                    "TransientXY",
+                    "MagnetostaticXY",
+                    "EddyCurrentXY",
+                    "ElectrostaticXY",
+                    "DCConductionXY",
+                    "ACConductionXY",
+                ]:
```
Comment (SMoraisAnsys) [inline pyaedt/modeler/cad/Primitives2D.py:315]:
```suggestion
                if self._app.solution_type in SOLUTION_TYPES:
```

File: pyaedt/modeler/cad/Primitives2D.py [hunk 4]
```diff
@@ -304,9 +305,29 @@ def create_region(self, pad_percent=300, is_percentage=True):
...
             if self._app.design_type == "2D Extractor" or self._app.design_type == "Maxwell 2D":
-                pad_percent = [pad_percent[0], pad_percent[1], 0, pad_percent[2], pad_percent[3], 0]
+                if self._app.solution_type in [
+                    "TransientXY",
+                    "MagnetostaticXY",
+                    "EddyCurrentXY",
+                    "ElectrostaticXY",
+                    "DCConductionXY",
+                    "ACConductionXY",
+                ]:
```
Comment (SMoraisAnsys) [inline pyaedt/modeler/cad/Primitives2D.py:328]:
```suggestion
                if self._app.solution_type in SOLUTION_TYPES:
```

**Review by SMoraisAnsys — CHANGES_REQUESTED** [review]

File: pyaedt/modeler/cad/Primitives2D.py (hunk 4 above)
Comment (SMoraisAnsys) [reply pyaedt/modeler/cad/Primitives2D.py:328]: @IreneWoyna Some minor changes to ease maintenance

**Comment by Samuelopez-ansys** [comment]
@SMoraisAnsys I think the suggestions are already merged.

File: pyaedt/modeler/cad/Primitives2D.py [hunk 5]
```diff
@@ -288,7 +288,8 @@ def create_region(self, pad_percent=300, is_percentage=True):
         ----------
         pad_percent : float, str, list of floats or list of str, optional
             Same padding is applied if not a list. The default is ``300``.
-            If a list of floats or str, interpret as adding for ``["+X", "+Y", "-X", "-Y"]``.
+            If a list of floats or str, interpret as adding ``["+X", "+Y", "-X", "-Y"]`` for XY geometry mode,
```
Comment (PipKat) [inline pyaedt/modeler/cad/Primitives2D.py:291]:
```suggestion
            If a list of floats or strings, interpret as adding ``["+X", "+Y", "-X", "-Y"]`` for XY geometry mode,
```
Avoid abbreviations in descriptions.

File: pyaedt/modeler/cad/Primitives2D.py [hunk 6]
```diff
@@ -288,7 +288,8 @@ def create_region(self, pad_percent=300, is_percentage=True):
...
+            and ``["+R", "+Z", "-Z"]`` for RZ geometry mode.
         is_percentage : bool, optional
             Region definition in percentage or absolute value. The default is `True``.
```
Comment (PipKat) [inline pyaedt/modeler/cad/Primitives2D.py:294]:
```suggestion
            Whether the region definition is a percentage or absolute value. The default is `True``.
```
Does "True" mean one or the other? If so, "The default is ``True``, in which case the region definition is a X."

**Review by SMoraisAnsys — APPROVED** [review]

File: _unittest/test_09_Primitives2D.py [hunk 7]
```diff
@@ -74,6 +81,25 @@ def test_06_create_region(self):
         region = self.aedtapp.modeler.create_region([100, 100, 100, 100, 100, 100])
         assert not region
 
+    def test_06_a_create_region_Z(self):
+        if self.axisymmetrical.modeler["Region"]:
+            self.axisymmetrical.modeler.delete("Region")
+        assert "Region" not in self.axisymmetrical.modeler.object_names
+        assert self.axisymmetrical.modeler.create_region([100, 50, 20])
```
Comment (gmalinve) [inline _unittest/test_09_Primitives2D.py:88]: @IreneWoyna I'd assert after each create_region that your Region object is now in the list of the objects. Before the delete()

**Review by gmalinve — APPROVED** [review]

**Pull request merged**